### PR TITLE
Use docker client in the integration tests

### PR DESCRIPTION
**Pull request by dmage**
/assign @legionus

### CODE REVIEW

File: hack/lib/constants.sh [hunk 1]
```diff
@@ -7,7 +7,7 @@ readonly OS_GO_PACKAGE=github.com/openshift/image-registry
 readonly OS_BUILD_ENV_GOLANG="${OS_BUILD_ENV_GOLANG:-1.8}"
 readonly OS_BUILD_ENV_IMAGE="${OS_BUILD_ENV_IMAGE:-openshift/origin-release:golang-${OS_BUILD_ENV_GOLANG}}"
 readonly OS_REQUIRED_GO_VERSION="go1.8"
-readonly OS_BUILD_ENV_WORKINGDIR="/go/${OS_GO_PACKAGE}"
+readonly OS_BUILD_ENV_WORKINGDIR="/go/src/${OS_GO_PACKAGE}"
```
Comment (dmage) [inline hack/lib/constants.sh]: @stevekuznetsov @kargakis ptal

Comment (stevekuznetsov) [reply hack/lib/constants.sh]: Why did we need this? /cc @smarterclayton

Comment (stevekuznetsov) [reply hack/lib/constants.sh]: IF we all need this, it should get percolated up

Comment (dmage) [reply hack/lib/constants.sh]: We need this because we want our package in GOPATH.

Comment (stevekuznetsov) [reply hack/lib/constants.sh]: Can you please push this to the `openshift/release` example? That is authoritative.

**Pull request merged**